### PR TITLE
Logrotate and S3 upload tweaks

### PR DESCRIPTION
**Pull request by tpetr**
- support file extensions for extra logrotated files
- support uploading files to a separate S3 bucket

### CODE REVIEW

File: SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorLogrotateAdditionalFile.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class SingularityExecutorLogrotateAdditionalFile {
+    private final String filename;
+    private final Optional<String> extension;
+    private final Optional<String> dateformat;
+
+    public static SingularityExecutorLogrotateAdditionalFile fromString(String value) {
```
Comment (ssalinas) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorLogrotateAdditionalFile.java]: Need an `@JsonCreator` for this one?

Comment (tpetr) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorLogrotateAdditionalFile.java]: Dang -- I forgot the annotation but Jackson is magically making it work (based on the method name & signature i assume). I'll add the annotation.

File: SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java [hunk 2]
```diff
@@ -58,7 +56,15 @@ private boolean writeS3MetadataFileForRotatedFiles(boolean finished) {
     final Path serviceLogParent = serviceLogOutPath.getParent();
     final Path logrotateDirectory = serviceLogParent.resolve(configuration.getLogrotateToDirectory());
 
-    return writeS3MetadataFile("default", logrotateDirectory, getS3GlobForLogRotatedFiles(), finished);
+    boolean result = writeS3MetadataFile("default", logrotateDirectory, String.format("%s*.gz*", taskDefinition.getServiceLogOutPath().getFileName()), Optional.<String>absent(), Optional.<String>absent(), finished);
+
+    int index = 1;
+    for (SingularityExecutorS3UploaderAdditionalFile additionalFile : configuration.getS3UploaderAdditionalFiles()) {
+      result = result && writeS3MetadataFile(additionalFile.getS3UploaderFilenameHint().or(String.format("extra%d", index)), logrotateDirectory, String.format("%s*.gz*", additionalFile.getFilename()), additionalFile.getS3UploaderBucket(), additionalFile.getS3UploaderKeyPattern(), finished);
```
Comment (ssalinas) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java]: I think this is forcing that all files to upload are gzipped, do we want to enforce that if it's possible to use this an a folder for uploading arbitrary files?

Comment (tpetr) [inline SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java]: Yep (and that's how it was operating before this PR: https://github.com/HubSpot/Singularity/pull/881/files#diff-e8314245f66e3ac2fc51f17320d6f1f6L197). Arbitrary file upload via a "drop" folder will come in a separate PR, if we attempt it

**Pull request merged**